pkg/utils/progressbar/progressbar_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package progressbar
+
+import (
+	"net/http"
+	"testing"
+)
+
+// TestNewTransport verifies the NewTransport function.
+func TestNewTransport(t *testing.T) {
+	// Mock base RoundTripper
+	base := &mockRoundTripper{}
+
+	// Create new transport with progress bar
+	transport := NewTransport(base)
+
+	// Verify that transport is an http.RoundTripper
+	// We can directly use it as an http.RoundTripper
+	resp, err := transport.RoundTrip(nil) // Perform a RoundTrip operation to ensure transport is functional
+	if err != nil {
+		t.Errorf("NewTransport did not return a functional RoundTripper: %v", err)
+	}
+	// Ensure the response body is closed
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			if err := resp.Body.Close(); err != nil {
+				t.Errorf("Error closing response body: %v", err)
+			}
+		}
+	}()
+}
+
+// TestTransportRoundTrip verifies the RoundTrip method of the transport.
+func TestTransportRoundTrip(t *testing.T) {
+	// Mock base RoundTripper
+	base := &mockRoundTripper{}
+
+	// Create new transport with progress bar
+	transport := NewTransport(base)
+
+	// Create a sample HTTP request
+	req, err := http.NewRequest(http.MethodGet, "https://example.com", nil)
```

Comment:
import "net/http/httptest" and use `httptest.NewServer()` to create a test server

Reply:
Okay, thanks!